examples/testmonitor/results/Results.cs

```diff
@@ -0,0 +1,180 @@
+        private static (double power, List<NamedValue> inputs, List<NamedValue> outputs) MeasurePower(int current, int voltage = 0)
+        {
+            var random = new Random();
+            var currentLoss = 1 - random.NextDouble() * 0.25;
+            var voltageLoss = 1 - random.NextDouble() * 0.25;
+            var power = current * currentLoss * voltage * voltageLoss;
+
+            // Record electrical current and voltage as inputs
+            var inputs = new List<NamedValue>()
+            {
+                new NamedValue("current", current),
+                new NamedValue("voltage", voltage)
+            };
+
+            // Record electrical power as an output
+            var outputs = new List<NamedValue>()
+            {
+                new NamedValue("power", power)
+            };
+
+            return (power, inputs, outputs);
+        }
+
+        /// <summary>
+        /// Builds a Test Monitor measurement parameter object for the power test.
+        /// </summary>
+        /// <param name="power">The electrical power measurement.</param>
+        /// <param name="lowLimit">The value of the low limit for the test.</param>
+        /// <param name="highLimit">The value of the high limit for the test.</param>
+        /// <param name="status">The measurement's pass/fail status.</param>
+        /// <returns>A list of test measurement parameters.</returns>
+        private static List<Dictionary<string, string>> BuildPowerMeasurementParams(double power, double lowLimit, double highLimit, Status status)
+        {
+            var parameter = new Dictionary<string, string>()
+            {
+                ["name"] = $"Power Test",
+                ["status"] = status.StatusType.ToString(),
+                ["measurement"] = $"{power}",
+                ["units"] = "Watts",
+                ["nominalValue"] = null,
+                ["lowLimit"] = $"{lowLimit}",
+                ["highLimit"] = $"{highLimit}",
+                ["comparisonType"] = "GELE"
+            };
+
+            var parameters = new List<Dictionary<String, String>>() { parameter };
+            return parameters;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StepData"/> object and
+        /// populates it to match the TestStand data model.
+        /// </summary>
+        /// <param name="name">The test step's name.</param>
+        /// <param name="stepType">The test step's type.</param>
+        /// <param name="inputs">The test step's input values.</param>
+        /// <param name="outputs">The test step's output values.</param>
+        /// <param name="parameters">The measurement parameters.</param>
+        /// <returns>The <see cref="StepData"/> used to create a test step.</returns>
+        private static StepData GenerateStepData(string name, string stepType, List<NamedValue> inputs, List<NamedValue> outputs, List<Dictionary<String, String>> parameters, Status status)
```

Comment:
Would it make sense to have most of these parameters have a default value of null so that callers can leave them out?